userspace/libscap/engine/modern_bpf/CMakeLists.txt

```diff
@@ -26,29 +26,46 @@ endif()
 # `USE_BUNDLED_MODERN_BPF` is `OFF`
 set(UNIQUE_BPF_O_FILE_NAME bpf_probe)
 
-# This must be a dir because we use it as an include path
-if(NOT MODERN_BPF_SKEL_DIR)
-	# Directory in which the BPF skeleton will be built
+# These must be directories, because we use them as include paths.
+if(NOT MODERN_BPF_SKEL_DIR AND NOT MODERN_BPF_EVENTS_DIMENSIONS_FILE_DIR)
+	# Directories in which the BPF skeleton and the events dimensions file will be built.
 	set(MODERN_BPF_SKEL_DIR "${CMAKE_BINARY_DIR}/skel_dir")
+	set(MODERN_BPF_EVENTS_DIMENSIONS_FILE_DIR "${CMAKE_BINARY_DIR}/ed_dir")
```

Comment:
to be honest I would not treat the dimension file like the skeleton. Loading the skeleton from a different folder is useful because you can build it with a recent toolchain on another machine and just provide scap with the `.h`. On the other side building the event dimension file in a separate step and provide it as an external file doesn't seem to have a real use case. I would keep the file path static and probably don't touch this cmake at all, WDYT?

Reply:
Yes agree, good suggestion!